include/kdbush.hpp

```diff
@@ -1,64 +1,89 @@
+// https://github.com/mourner/kdbush.hpp
+
 #pragma once
 
 #include <algorithm>
 #include <cmath>
 #include <vector>
 
+template<class TFloat = double, class TIndex = std::size_t>
```

Comment:
> What are the reasons to template std::size_t as well?

Memory waste on x64 platforms, and the resulting cache trashing. Granted, I haven't performed any measurements, but I have experienced (meaning; measured and fixed in production code) the difference (for example) between using arrays of 8 byte pointers (on x64) versus arrays of 32 or 16 bit indices. Incidentally, my use-case never has more than 20.000 or so points in the bush, so I'm using `uint16_t`.

In fact, if I had to pick only one (which is kind of silly), I would choose to template the index type before I'd template the coordinate type. At least when using doubles over floats you're getting some precision in return that can be useful. But using size_t on 64-bit platforms when I only stick 10.000 points in there, that's just wasting perfectly valid bits... :-)

A silly choice of course, since it's rather easy to template over both.